static/js/stream_list.js

```diff
@@ -529,29 +540,143 @@ exports.clear_and_hide_search = function () {
 };
 
 function focus_stream_filter(e) {
+    if ($('#stream_filters li.narrow-filter.highlighted_stream').length === 0) {
+        // Highlight
+        var all_streams = $('#stream_filters li.narrow-filter');
+        exports.highlight_first(all_streams, 'highlighted_stream');
+    }
     e.stopPropagation();
 }
 
-function maybe_select_stream(e) {
-    if (e.keyCode === 13) {
-        // Enter key was pressed
+function focusout_stream_filter() {
+    // Undo highlighting
+    $('#stream_filters li.narrow-filter.highlighted_stream').removeClass('highlighted_stream');
+}
+
+function keydown_enter_key() {
+    // Is there at least one stream?
+    if ($('#stream_filters li.narrow-filter').length > 0) {
+
+        var selected_stream_id = $('#stream_filters li.narrow-filter.highlighted_stream')
+                                    .expectOne().data('stream-id');
+
+        var top_stream = stream_data.get_sub_by_id(selected_stream_id);
+
+        if (overlays.is_active()) {
+            ui_util.change_tab_to('#home');
+        }
+        exports.clear_and_hide_search();
+        narrow.by('stream', top_stream.name,
+            {select_first_unread: true, trigger: 'sidebar enter key'});
+    }
```

Comment:
If there's no stream, should the behavior here still capture the `enter` keypress?  Just wondering if this function should return `handled` to the caller.

Reply:
Normally, the ```enter``` press needs to be suppressed to avoid a line-break in the compose box. If there is no stream, the suppression of the ```enter``` press guarantees that no line-break character is passed to the ```<input>```. I think the ```<input>``` cannot receive a line-break character, so the suppression is probably redundant but I would leave it like that.
I added a clarifying comment in line 653.